runtime/standalone-shared/src/main/java/eu/maveniverse/maven/mima/runtime/shared/internal/SettingsUtils.java

```diff
@@ -0,0 +1,187 @@
+            org.apache.maven.settings.ActivationFile settingsFile = settingsActivation.getFile();
+            if (settingsFile != null) {
+                ActivationFile file = new ActivationFile();
+                file.setExists(settingsFile.getExists());
+                file.setMissing(settingsFile.getMissing());
+                activation.setFile(file);
+            }
+            profile.setActivation(activation);
+        }
+        profile.setProperties(settingsProfile.getProperties());
+        List<Repository> repos = settingsProfile.getRepositories();
+        if (repos != null) {
+            for (Repository repo : repos) {
+                profile.addRepository(convertFromSettingsRepository(repo));
+            }
+        }
+        List<Repository> pluginRepos = settingsProfile.getPluginRepositories();
+        if (pluginRepos != null) {
+            for (Repository pluginRepo : pluginRepos) {
+                profile.addPluginRepository(convertFromSettingsRepository(pluginRepo));
+            }
+        }
+        return profile;
+    }
+
+    private static org.apache.maven.model.Repository convertFromSettingsRepository(Repository settingsRepo) {
+        org.apache.maven.model.Repository repo = new org.apache.maven.model.Repository();
+        repo.setId(settingsRepo.getId());
+        repo.setLayout(settingsRepo.getLayout());
+        repo.setName(settingsRepo.getName());
+        repo.setUrl(settingsRepo.getUrl());
+        if (settingsRepo.getSnapshots() != null) {
+            repo.setSnapshots(convertRepositoryPolicy(settingsRepo.getSnapshots()));
+        }
+        if (settingsRepo.getReleases() != null) {
+            repo.setReleases(convertRepositoryPolicy(settingsRepo.getReleases()));
+        }
+        return repo;
+    }
+
+    private static org.apache.maven.model.RepositoryPolicy convertRepositoryPolicy(RepositoryPolicy settingsPolicy) {
+        org.apache.maven.model.RepositoryPolicy policy = new org.apache.maven.model.RepositoryPolicy();
+        policy.setEnabled(settingsPolicy.isEnabled());
+        policy.setUpdatePolicy(settingsPolicy.getUpdatePolicy());
+        policy.setChecksumPolicy(settingsPolicy.getChecksumPolicy());
+        return policy;
+    }
+
+    private static Repository convertToSettingsRepository(org.apache.maven.model.Repository modelRepo) {
+        Repository repo = new Repository();
+        repo.setId(modelRepo.getId());
+        repo.setLayout(modelRepo.getLayout());
+        repo.setName(modelRepo.getName());
+        repo.setUrl(modelRepo.getUrl());
+        if (modelRepo.getSnapshots() != null) {
+            repo.setSnapshots(convertRepositoryPolicy(modelRepo.getSnapshots()));
+        }
+        if (modelRepo.getReleases() != null) {
+            repo.setReleases(convertRepositoryPolicy(modelRepo.getReleases()));
+        }
+        return repo;
+    }
+
+    private static RepositoryPolicy convertRepositoryPolicy(org.apache.maven.model.RepositoryPolicy modelPolicy) {
```

Comment:
It would probably be good to align also with convertTo/convertFrom names to avoid confusion.

Reply:
I thought even to inline this into support class, as copied class from Maven is already "mutilated" (unused methods removed, reformatted)